src/mbgl/style/sources/geojson_source_impl.cpp

```diff
@@ -13,19 +13,12 @@
 namespace mbgl {
 namespace style {
 
-class GeoJSONVTData : public GeoJSONData, public std::enable_shared_from_this<GeoJSONVTData> {
+class GeoJSONVTData : public GeoJSONData {
 public:
     void getTile(const CanonicalTileID& id, const std::function<void(TileFeatures)>& fn) final {
         assert(fn);
-        std::weak_ptr<GeoJSONVTData> weak = shared_from_this();
         scheduler->scheduleAndReplyValue(
-            [id, weak, this]() -> TileFeatures {
-                if (auto self = weak.lock()) {
-                    return impl.getTile(id.z, id.x, id.y).features;
-                }
-                return {};
-            },
-            fn);
+            [id, impl = this->impl]() -> TileFeatures { return impl->getTile(id.z, id.x, id.y).features; }, fn);
```

Comment:
nit: I think fn can be moved.
`void GeoJSONData::getTile(const CanonicalTileID& id, std::function<void(TileFeatures)> fn) = 0`?

Reply:
It's passed as const ref..